### ISSUE TITLE
scAlignOptions Question

### ISSUE THREAD

**Issue by tbrunetti** [quon-titative-biology/examples#3]
I am working on running scAlign and I have some questions about how to determine what numeric value to set some of the option values?

steps=15000  -- How can I determine what is the best number of iteration/step to use?
batch.size=300 -- Is batch size dictated by total samples in the dataset after merging?  Should I train on about 10% of the total samples or how did you come up with the number 300?
perplexity=30 -- How did you come up with a perplexity of 30?  Is there any analysis I can do to determine what this number should be?
norm=TRUE 
batch.norm.layer=FALSE
architecture="large" - How do I determine which architecture size I should use?
num.dim=64 - How do I determine how many dimensions I should be using?

Any advice would be greatly appreciated!

Thanks

**Comment by UCDNJJ** [comment]
Hi @tbrunetti,

The default parameters were set to be generally applicable to most alignment problems for scRNAseq. Here's a few insights on how to refine each of the parameters for your specific use case:

steps: Should be determined based on convergence of the loss function which is printed every 100 steps by default. Once the value stops decreasing then you've reached an optimal step, you can determine the convergence point by running scAlign with a large step value initially.

batch.size: Right, this is based on the total samples after merging. 

perplexity: Effectively determines the number of neighbors for each sample in the mini-batch, the bandwidth of the Gaussian kernel (similar to t-SNE). We set perplexity to 30 as that is default value also used by tSNE. The smaller perplexity the more local structure will be captured.

To determine the optimal value for batch.size and perplexity you would have to perform a grid search across multiple values for each parameter and compare either the final loss values or use alignment score as an evaluation metric. A good starting point would be batch.size=5-10% of your data, and perplexity = 10% of batch.size. 

norm and batch.norm.layer: Both perform normalizing operations prior and during training respectively, usually best to leave these on TRUE. 

architecture: the larger the network architecture the more power scAlign has to warp the data into alignment. It would be best to choose the smallest and most regularized model which produces an acceptable alignment. I would start with small and work up to large.

num.dim: We found that 64 works well across all of our experiments, if you would like to optimize this parameter then comparing loss values and alignment scores would be the best way to do so.

Hopefully this helps. We are planning to update scAlign for Tensorflow2 which has built-in early stopping and makes parameter optimization a bit easier.

**Comment by tbrunetti** [comment]
This is great and very, very helpful!  Thank you for your thorough response and very quick reply!
